test/table_test.rb

```diff
@@ -137,7 +141,47 @@ def test_transform_line
   end
 
   def test_encode_with
-    # encode_with(coder)
+    table = NdrImport::Table.new
+    assert table.instance_variables.include?(:@row_index)
+    refute table.class.all_valid_options.include?('row_index')
+    assert_nil table.columns
+
+    coder = {}
+    table.encode_with(coder)
+    assert coder.has_key?('columns')
+
+    yaml_output = table.to_yaml
+    assert yaml_output.include?('columns')
+    refute yaml_output.include?('row_index')
+  end
```

Comment:
It is worth adding some assertions that `YAML.load(ndr_table.to_yaml)` gives you back an `NdrTable` with the options configured as expected? This PR isn't defining `init_with` alongside `encode_with` (which is fine), but might be worth checking that the serialisation this emits can be reversed!